source/common/protobuf/protobuf.h

```diff
@@ -27,6 +27,14 @@
 #include "google/protobuf/util/type_resolver_util.h"
 #include "google/protobuf/wrappers.pb.h"
 
+#ifdef WIN32
+// one of the above headers includes <windows.h>, so undef some interfering symbols
+#undef TRUE
+#undef DELETE
+#undef ERROR
+#undef GetMessage
```

Comment:
This seems pretty fragile, is there some best practice for importing Protobuf headers on Windows or other projects we can crib from?

Reply:
I'll attach the explanation top-post in the review comments, but I agree we should consume our own platform.h here and delegate inclusion of windows.h at that inflection point, trashing MS macros at the time we include windows.h. Individual modules should never be injecting MS SDK API headers anywhere without coordination.

Reply:
Thanks, I like this solution.